packages/vitest/src/api/setup.ts

```diff
@@ -88,13 +93,11 @@ export function setup(vitestOrWorkspace: Vitest | WorkspaceProject, server?: Vit
           return fs.writeFile(id, content, 'utf-8')
         },
         async saveSnapshotFile(id, content) {
-          if (!ctx.snapshot.resolvedPaths.has(id))
-            throw new Error(`Snapshot file "${id}" does not exist.`)
           await fs.mkdir(dirname(id), { recursive: true })
           return fs.writeFile(id, content, 'utf-8')
         },
         async removeSnapshotFile(id) {
-          if (!ctx.snapshot.resolvedPaths.has(id) || !existsSync(id))
+          if (!existsSync(id))
```

Comment:
This allows deleting any file on the file system

Reply:
Okay, I'll revert this, but I did have a question about `removeSnapshotFile()` - is this method used in practice? I couldn't really think of a use case and wondered if the code could be simplified (now or later) by removing it?

Reply:
Snapshot is removed when test file is deleted or there are no snapshots (`toMatchSnapshot` calls) in the test file anymore (watch mode)

Reply:
Okay understood. Applied the access check here.